migrations/2024-08-28-101632_create-progress-table/up.sql

```diff
@@ -0,0 +1,13 @@
+-- Your SQL goes here
+CREATE TABLE progress (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    user_id UUID NOT NULL REFERENCES users(id),
+    challenge_id UUID NOT NULL REFERENCES challenges(id),
+    status VARCHAR NOT NULL CHECK (status IN ('completed', 'in_progress', 'not_started')) DEFAULT 'not_started',
+    progress_details JSONB,
+    created_at TIMESTAMP NOT NULL DEFAULT now(),
+    updated_at TIMESTAMP NOT NULL DEFAULT now()
```

Comment:
This will be set to datetime at time of row creation and will need to be updated manually on every update after that (unless we add a database trigger).  Just something to note.